lib/strings.nix

```
@@ -252,11 +252,9 @@ rec {
       hasInfix "foo" "abcd"
       => false
   */
-  hasInfix = infix: content:
-    let
-      drop = x: substring 1 (stringLength x) x;
-    in hasPrefix infix content
-      || content != "" && hasInfix infix (drop content);
+  hasInfix = infix:
+    let matchInfix = builtins.match ".*${escapeRegex infix}.*";
+    in content: matchInfix content != null;
```

```suggestion
  hasInfix = infix: content:
    builtins.match ".*${escapeRegex infix}.*" content != null;
```
the let introduces an extra env and thunk that can be elided as well